cmake/sysbuild/suit.cmake

```diff
@@ -0,0 +1,347 @@
+#   'path' - path to resolve
+#   'relative_root' - root folder used in case of relative path
+#   'output_variable' - variable to store results
+function(suit_set_absolute_or_relative_path path relative_root output_variable)
+  if(NOT path MATCHES "^/.*" AND NOT path MATCHES "^[a-zA-Z]:/.*")
+    file(REAL_PATH "${relative_root}/${path}" path)
+  endif()
+  set(${output_variable} "${path}" PARENT_SCOPE)
+endfunction()
+
+# Sign an envelope using SIGN_SCRIPT.
+#
+# Usage:
+#   suit_sign_envelope(<input_file> <output_file>)
+#
+# Parameters:
+#   'input_file' - path to input unsigned envelope
+#   'output_file' - path to output signed envelope
+function(suit_sign_envelope input_file output_file)
+  cmake_path(GET SYSBUILD_NRF_MODULE_DIR PARENT_PATH NRF_DIR_PARENT)
+  sysbuild_get(CONFIG_SUIT_ENVELOPE_SIGN_SCRIPT IMAGE ${DEFAULT_IMAGE} VAR CONFIG_SUIT_ENVELOPE_SIGN_SCRIPT KCONFIG)
+  suit_set_absolute_or_relative_path(${CONFIG_SUIT_ENVELOPE_SIGN_SCRIPT} ${NRF_DIR_PARENT} SIGN_SCRIPT)
+  if(NOT EXISTS ${SIGN_SCRIPT})
+    message(FATAL_ERROR "DFU: ${CONFIG_SUIT_ENVELOPE_SIGN_SCRIPT} does not exist. Corrupted configuration?")
+  endif()
+  set_property(
+    GLOBAL APPEND PROPERTY SUIT_POST_BUILD_COMMANDS
+    COMMAND ${PYTHON_EXECUTABLE} ${SIGN_SCRIPT}
+    --input-file ${input_file}
+    --output-file ${output_file}
+  )
+endfunction()
+
+# Render jinja templates using passed arguments.
+# Function uses core_arguments which is list of folowing entries:
+#   --core <target_name>,<locatino_of_firmware_binary_file>,<location_of_edt_file_representing_dts>
+#
+# Usage:
+#   suit_render_template(<input_file> <output_file> <core_arguments>)
+#
+# Parameters:
+#   'input_file' - path to input jinja template
+#   'output_file' - path to output yaml file
+#   'core_arguments' - list of arguments for registered cores
+function(suit_render_template input_file output_file core_arguments)
+  set(TEMPLATE_ARGS)
+  list(APPEND TEMPLATE_ARGS
+    --template-suit ${input_file}
+    --output-suit ${output_file}
+    --zephyr-base ${ZEPHYR_BASE}
+  )
+  list(APPEND TEMPLATE_ARGS ${core_arguments})
+  list(APPEND TEMPLATE_ARGS --artifacts-folder "${SUIT_ROOT_DIRECTORY}")
+   set_property(
+    GLOBAL APPEND PROPERTY SUIT_POST_BUILD_COMMANDS
+    COMMAND ${PYTHON_EXECUTABLE} ${SUIT_GENERATOR_BUILD_SCRIPT}
+    template
+    ${TEMPLATE_ARGS}
+  )
+endfunction()
+
+# Create binary envelope from input yaml file.
+#
+# Usage:
+#   suit_create_envelope(<input_file> <output_file>)
+#
+# Parameters:
+#   'input_file' - path to input yaml configuration
+#   'output_file' - path to output binary suit envelope
+function(suit_create_envelope input_file output_file)
+  set_property(
+    GLOBAL APPEND PROPERTY SUIT_POST_BUILD_COMMANDS
+    COMMAND ${PYTHON_EXECUTABLE} ${SUIT_GENERATOR_CLI_SCRIPT}
+    create
+    --input-file ${input_file}
+    --output-file ${output_file}
+  )
+  sysbuild_get(CONFIG_SUIT_ENVELOPE_SIGN IMAGE ${DEFAULT_IMAGE} VAR CONFIG_SUIT_ENVELOPE_SIGN KCONFIG)
```

Comment:
Could we instead add a "boolean" argument to this function stating if we should sign/not sing the envelope and do the extracting of the `CONFIG_SUIT_ENVELOPE_SIGN` outside of this function?

The reason for that is that the secdom/sysctrl build systems would need to use this function and they do not use sysbuild (and do not have knowledge of the `sysbuild_get` function

Reply:
Done